Makefile

```diff
@@ -1,88 +1,114 @@
 testable_packages=$(shell go list ./... | egrep -v 'constants|mocks|testing')
 project=$(shell basename $(PWD))
-project_test=${project}-test
-pg_dep=$(project)_postgres_1
-test_packages=`find . -type d -name "docker_data" -prune -o \
-							-type f -name "*.go" ! \( -path "*vendor*" \) -print \
-							| sed -En "s/([^\.])\/.*/\1/p" | uniq`
-database=postgres://postgres:$(project)@localhost:8432/$(project)?sslmode=disable
-database_test=postgres://postgres:$(project)@localhost:8432/$(project_test)?sslmode=disable
-platform=darwin
+test_db_name=${project}-test
+pg_docker_image=$(project)_postgres_1
+db_url=postgres://postgres:$(project)@localhost:8432/$(project)?sslmode=disable
+db_test_url=postgres://postgres:$(project)@localhost:8432/$(test_db_name)?sslmode=disable
+uname_S=$(shell uname -s)
+
+# TravisCI runs in Linux instances, while developers run in MacOS machines
+ifeq ($(uname_S), Darwin)
+  platform := darwin
+else
+  platform := linux
+endif
 
 export GO111MODULE=on
 
-setup: setup-project setup-deps
+.PHONY: all
+all: setup-migrate download-mod db-setup
 
-setup-project:
-	@go mod download
+.PHONY: all-ci
+all-ci: setup-migrate download-mod db-setup-test
 
-setup-deps:
-	@make deps
-	@make migrate
+.PHONY: build
+build:
+	@mkdir -p bin && go build -o ./bin/$(project) .
 
-# run this if you don't have migrate
-setup-migrate:
-	@curl -L https://github.com/golang-migrate/migrate/releases/download/v4.4.0/migrate.$(platform)-amd64.tar.gz | tar xvz
-	@mv migrate.$(platform)-amd64 /usr/local/bin/migrate
+.PHONY: build-docker
+build-docker:
+	@docker build -t $(project) .
 
-deps:
-	@mkdir -p docker_data && docker-compose up -d postgres
-	@until docker exec $(pg_dep) pg_isready; do echo 'Waiting Postgres...' && sleep 1; done
-	@docker exec $(pg_dep) createuser -s -U postgres $(project) 2>/dev/null || true
-	@docker exec $(pg_dep) createdb -U $(project) $(project) 2>/dev/null || true
+.PHONY: run
+run:
+	@reflex -c reflex.conf -- sh -c ./bin/Will.IAM start-api
 
-deps-test:
-	@mkdir -p docker_data && docker-compose up -d postgres
-	@until docker exec $(pg_dep) pg_isready; do echo 'Waiting Postgres...' && sleep 1; done
-	@docker exec $(pg_dep) createuser -s -U postgres $(project) 2>/dev/null || true
-	@docker exec $(pg_dep) createdb -U $(project) $(project_test) 2>/dev/null || true
-	@make migrate-test
+.PHONY: test
+test: db-setup-test test-unit test-integration db-stop-test
 
-stop-deps:
+# Installs the golang-migrate dependency if its not already installed.
+.PHONY: setup-migrate
+setup-migrate:
+	@echo "Platform: ${platform}"
+ifeq ($(shell command -v migrate),)
+	@echo "Installing migrate..."
```

Comment:
FYI, most of this output would be unnecessary if we weren't silencing all commands.